platform/stm32f4xx/export/platform/spi_bus.hpp

```diff
@@ -840,6 +903,49 @@ ecl::err spi_i2s_bus< spi_i2s_config >::i2s_set_data_format_private(uint32_t val
     return ecl::err::ok;
 }
 
+template< class spi_i2s_config >
+template< spi_bus_mode mode >
+std::enable_if_t< mode == spi_bus_mode::DMA, ecl::err >
+spi_i2s_bus< spi_i2s_config >::enable_circular_mode()
+{
+    // Cannot enable circular mode if transfer is ongoing
+    if ( !(m_status & tx_complete) || !(m_status & rx_complete)) {
```

Comment:
Also, I think this is clearly a programming error if client wish to start circular mode when xfer was not handled yet. Let's place assert here